src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1484,13 +1489,9 @@ static void perform_stream_op_locked(void* stream_op,
     on_complete->next_data.scratch |= CLOSURE_BARRIER_MAY_COVER_WRITE;
     s->fetching_send_message_finished = add_closure_barrier(op->on_complete);
     if (s->write_closed) {
-      op->payload->send_message.stream_write_closed = true;
```

Comment:
so is the high-level bit of the PR that stream_write_closed is being replaced by GRPC_ERROR_EOS or are there subtle differences?

Reply:
The main thing is to fix it for inproc transport. I just thought that using a special error message might be better